admin_guide/image_pull_secrets.adoc

```diff
@@ -0,0 +1,46 @@
+= Image Pull Secrets
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+Docker registries can be secured to prevent unauthorized parties from accessing certain images.  If you are using the OpenShift docker registry and are pulling images for pods located in the same project, then your service account should already have permissions and no additional action should be required.  If this is not the case, then additional configuration steps are required.
+
+=== OpenShift Docker Registry Authentication and Authorization
+The OpenShift Docker Registry authenticates using the same tokens as the OpenShift API.  To perform a `docker login` against the OpenShift Docker registry, you can choose any username and email, but the password must be a valid OpenShift token.  In order to pull an image, the authenticated user must have `get` rights on the requested `imagestreams/layer`.  In order to push an image, the authenticated user must have `update` rights on the requested `imagestreams/layer`.
+
+By default, all service accounts in a project have rights to pull any image in the same project and the builder service account has rights to push any image in the same project.
+
+=== Allowing Pods in ProjectA to reference images from ProjectB
+In order to allow pods in ProjectA to reference images in ProjectB, a service account in ProjectA should be bound to the `system:image-puller` role in ProjectB (`oc policy add-role-to-user serviceaccount:ProjectA:default --namespace=ProjectB`).  After adding that role, the pods in ProjectA that reference the default service account will be able to pull images from Project B.
+
+=== Allowing Pods in ProjectA to reference images from secured Docker registry
+If you are trying to pull a secured Docker image, but **not** using the OpenShift Docker registry then you must create a dockercfg secret and add it to your service account.
+
+If you already have a .dockercfg file, you can create a secret from that file by running
+```
+oc secrets new my-pull-secret .dockercfg=path/to/my/.dockercfg
+```
+
+If you don't already have a .dockercfg file, you can create a secret by running
+```
+oc secrets new-dockercfg my-pull-secret --docker-server=DOCKER_REGISTRY_SERVER --docker-username=DOCKER_USER --docker-password=DOCKER_PASSWORD --docker-email=DOCKER_EMAIL
+```
+
+Once you've created your secret, you need to add that secret to your service account.
+
+If you are going to use your secret for pulling images for pods, then you can add the secret to your service account by running
+```
+oc secrets add serviceaccount/default secrets/my-pull-secret --for=pull
+```
+
+If you are going to use your secret for push and pulling build images, then the secret needs to be mountable inside of a pod.  You can do this by running
+```
+oc secrets add serviceaccount/default secrets/my-pull-secret
```

Comment:
should this be the `builder` service account, if the example is for running builds?